### PR TITLE
Compile action for pull requests

### LINKED ISSUE #708: Compile action for pull requests
Add an action on pull requests **only** that run `npm run compile` on changes to anything in the src directory. This would make contributing more efficient as so many people forget to compile their work on PRs and time gets wasted in review to request the compilation.

### PR DESCRIPTION
**Pull request by varunks99**
Issue: This resolves #708 

Added an action that runs `npm run compile` whenever there is a pull request with changes to the `src` directory.

@shahednasser

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Compile source code
+
+on:
+  pull_request:
+    branches:
```
Comment (shahednasser) [inline .github/workflows/compile.yml:5]: Please remove the branches restriction, as a lot of contributors use their own branch

**Review by shahednasser — CHANGES_REQUESTED** [review]
Great work! I just have one comment below

File: .github/workflows/compile.yml (hunk 1 above)
Comment (varunks99) [reply .github/workflows/compile.yml]: Done

**Comment by shahednasser** [comment]
Great thank you!

@all-contributors please add @varunks99 for code

**Pull request merged**